src/Datagrid/ListMapper.php

```diff
@@ -96,15 +95,18 @@ public function addIdentifier($name, ?string $type = null, array $fieldDescripti
     }
 
     /**
-     * @param FieldDescriptionInterface|string $name
      * @param array<string, mixed>             $fieldDescriptionOptions
      *
      * @throws \LogicException
      *
      * @return static
      */
-    public function add($name, ?string $type = null, array $fieldDescriptionOptions = []): self
+    public function add(string $name, ?string $type = null, array $fieldDescriptionOptions = []): self
     {
+        if (isset($fieldDescriptionOptions['role']) && !$this->getAdmin()->isGranted($fieldDescriptionOptions['role'])) {
```

Comment:
In FormMapper this was an early check, I think it's a good idea to have an early return instead of doing all the work and not adding the field.